src/main/io/gps.c

```diff
@@ -2608,9 +2620,20 @@ void onGpsNewData(void)
 #ifdef USE_GPS_RESCUE
     gpsRescueNewGpsData();
 #endif
+
 #ifdef USE_GPS_LAP_TIMER
     gpsLapTimerNewGpsData();
 #endif // USE_GPS_LAP_TIMER
+
+}
+
+bool isNewGPSDataAvailable(void) {
```

Comment:
Each consumer needs it's own lastGpsStamp.
```suggestion
bool isNewGPSDataAvailable(uint16_t* stamp) {
```

and call it:
```
static uint16_t gpsStamp = ~0;
if (isNewGPSDataAvailable(&gpsStamp)) {

Reply:
Very good, thank you.